plugins/src/webrtcsink/imp.rs

```diff
@@ -2551,6 +2578,27 @@ impl ObjectImpl for WebRTCSink {
                     res
                 })
                 .build(),
+                /*
+                 * RsWebRTCSink::encoder-setup:
+                 * @consumer_id: Identifier of the consumer
+                 * @pad_name: The name of the corresponding input pad
+                 * @encoder: The constructed encoder
+                 *
+                 * This signal can be used to tweak @encoder properties.
+                 *
+                 * Returns: True if the encoder is entirely configured,
+                 * False if webrtcsink should apply a default configuration
+                 */
+                glib::subclass::Signal::builder(
+                    "encoder-setup",
+                    &[
+                        String::static_type().into(),
+                        String::static_type().into(),
+                        gst::Element::static_type().into(),
+                    ],
+                    bool::static_type().into(),
```

Comment:
You probably want to
  - Use an accumulator function that causes the next handler to not be called anymore if `false` (or `true`, your decision) is returned
  - Have a default class handler that does the default setup (i.e. the `configure_encoder()` function)

That way applications can connect signal handlers before or after the class handler, and can decide if they just want to extend the behaviour or don't want the class handler to be called. Just like many such signals work in GTK for example.